code/HabitTracker/app/src/main/java/com/example/habittracker/activities/fragments/AddEventFragment.java

```diff
@@ -260,7 +260,7 @@ public void onClick(DialogInterface dialogInterface, int i) {
 
                             editM.setStartDate(stringToArraylist(date.getText().toString()));
                             editM.setLocation(location1.getText().toString());
-                            editM.setHabit(attachedHabit);
+                            editM.setHabitInDB(attachedHabit, 1);
```

Comment:
If this function works correctly, you should put it in onOkPressed function, not here. Comment of a habit event will not be add to database if you put it here. We should call updateDB/editDB after all attributes being properly set up.

Reply:
As we discussed, I'll move it down for clarity.